src/compiler/types.ts

```diff
@@ -574,15 +574,15 @@ namespace ts {
         /* @internal */ InWithStatement               = 1 << 24, // If any ancestor of node was the `statement` of a WithStatement (not the `expression`)
         JsonFile                                      = 1 << 25, // If node was parsed in a Json
         /* @internal */ TypeCached                    = 1 << 26, // If a type was cached for node at any point
-        /* @internal */ InNonEmittingNode             = 1 << 27, // If a node was in a context where expressions are never emitted: type queries, abstract members, computed property names of types.
+        /* @internal */ InNonEmittingExpression       = 1 << 27, // If a node was in a context where expressions are never emitted: type queries, abstract members, computed property names of types.

```

Comment:
Can you tell me more about this? I'm not as familiar with this part of the codebase, and I'm not sure why we're changing it from Nodes to Expressions

Reply:
I wrote that flag as part of this PR, just changing the meaning between commits. But I might have to do something different, because apparently it is still super hard to tell whether we’re in a type position or not unless we’re in the parser. 🙄 

The problem with setting context flags is that they limit what the incremental parser will reuse, so I was hesitant to set this across large type declaration like interfaces and object literal types. We also don’t have a ton of test coverage for incremental parsing so it’s hard to tell how much impact that would have on a typical source file. So I’m now looking to see if I can keep setting this flag on all type nodes, but wire in some help for the incremental parser here and there. I’ve done a similar optimization with ambient nodes in the past, so I’m hopeful: https://github.com/microsoft/TypeScript/pull/32849